### PR TITLE
Find `@Option` annotations without `example` values

### PR DESCRIPTION
**Pull request by timtebeek**
As came up in Slack: https://github.com/openrewrite/gh-automation/issues/45#issuecomment-1891827588

### CODE REVIEW

**Comment by timtebeek** [comment]
@koppor / @MBoegers since you were involved; if this gets a thumbs up I'll merge and deploy. :)

File: rewrite-java-test/src/test/java/org/openrewrite/java/recipes/MissingOptionExampleTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                      return "Find `@Option` annotations that are missing `example` values.";
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void skipBoolean() {
```
Comment (koppor) [inline rewrite-java-test/src/test/java/org/openrewrite/java/recipes/MissingOptionExampleTest.java:105]: I don't understand the intention of this test

**Review by koppor — COMMENTED** [review]
Looks good. I miss some documentation for one test case

File: rewrite-java-test/src/test/java/org/openrewrite/java/recipes/MissingOptionExampleTest.java (hunk 1 above)
Comment (koppor) [reply rewrite-java-test/src/test/java/org/openrewrite/java/recipes/MissingOptionExampleTest.java:105]: Ah, I get it. Booleans don't need ab example.

**Review by koppor — APPROVED** [review]

File: rewrite-java-test/src/test/java/org/openrewrite/java/recipes/MissingOptionExampleTest.java (hunk 1 above)
Comment (timtebeek) [reply rewrite-java-test/src/test/java/org/openrewrite/java/recipes/MissingOptionExampleTest.java:105]: indeed, we already insert [`false` by default for booleans](https://github.com/openrewrite/rewrite-recipe-markdown-generator/blob/821ab474934d72d21a5ec5b956ea92b2c155759f/src/main/kotlin/org/openrewrite/RecipeMarkdownGenerator.kt#L1064-L1065).

**Comment by MBoegers** [comment]
nice! just out of curiositiy, wouldn't be a `//todo:` be easier to find?

**Comment by timtebeek** [comment]
Right now we produce a search marker, which has separate handling in the Moderne Platform to show the results there. That would be my main goal with this recipe to have a list of missing examples, and a hint at how to resolve that on a PR. Inserting a comment is also an option, but lacks that more sophisticated handling.

**Comment by MBoegers** [comment]
Yes, I understand that they are SearchResults but simply adding “todo” to the comment would make visible for InteliJs TODO tab and combined with a Issue link it would be possible to have an automated todo list. But it will also work this way 😉

Von: Tim te Beek ***@***.***>
Antworten an: openrewrite/rewrite ***@***.***>
Datum: Montag, 15. Januar 2024 um 20:27
An: openrewrite/rewrite ***@***.***>
Cc: "Bögershausen, Merlin" ***@***.***>, Mention ***@***.***>
Betreff: Re: [openrewrite/rewrite] Find ***@***.***` annotations without `example` values (PR #3917)


Right now we produce a search marker, which has separate handling in the Moderne Platform to show the results there. That would be my main goal with this recipe to have a list of missing examples, and a hint at how to resolve that on a PR. Inserting a comment is also an option, but lacks that more sophisticated handling.

**Comment by timtebeek** [comment]
Adding a link to the recipe as it can be run from the platform:
https://app.moderne.io/recipes/org.openrewrite.java.recipes.MissingOptionExample
From this initial run it appears we only have a handful of cases still to cover 🎉 
https://app.moderne.io/results/cCwlMe3wr

image

**Comment by timtebeek** [comment]
Looking over the results once more we can flag even fewer cases by ignoring primitives https://github.com/openrewrite/rewrite/commit/b044c5de771a6161d27d5c6170c2487ff5c47d90

**Pull request merged**